### DISCUSSION TITLE
Opt-in notifications for threads

### DISCUSSION THREAD

**Discussion by bmos** [element-hq/element-meta#984] [Ideas]
#### What would you like to do?
In many messaging clients like Discord, Slack, Mattermost, and others, joining a channel means you get notified of messages in that channel but not in threads within that channel unless you have participated in them. I would like to see this in Element.

#### Why would you like to do it?
It can be hard to follow everything that is going on in busy rooms, so it's often best not to try. By having opt-in notifications, you can avoid getting pinged by every little thing that's going in in a room.

#### How would you like to achieve it?
Reacting with emoji, replying in a thread, or reacting to a reply in a thread should subscribe you to the thread. A button should be included in the overflow menu to allow unsubscribing from the thread afterwards.

### Have you considered any alternatives?

Having everything on by default and simply allowing muting of threads could be another solution.

**Comment by MickVolt** [comment]
Yes, please!

If there would be an unsubscribe option, why not just add a subscribe option as well?
Then not everyone has to react (via emoji?) or reply (chat message?), to subscribe.

It would be good enough for me to not have these subscribe & unsubscribe options.

My question is: Why is this a discussion? How to make this a (high priority?) issue? And how to make this a reality?

**Reply by bmos** [reply]
I started this as an issue, but it was moved here by mods.
https://github.com/vector-im/element-meta/issues/983

**Comment by bartvdbraak** [comment]
We've received recurring feedback from our users regarding thread notifications:

> It seems like they're treated the same as regular channel messages. It would make more sense if participants in a thread could receive notifications.

Are there any updates or plans to enhance this experience in the near future?
